File: python/template.py
import math

# these are the tree and utility libraries I created
from tree import Tree
import util

# TODO: tweak this to do whatever you want as I described in the Overleaf
def next(x):
    return []


# here we create a tree with value 0 and no children and check it for 20
# iterations, but feel free to change that up
def main():
    tree = Tree(0, [])
    for i in range(20):
        print(tree.longest_path())
        tree.step(next)
    print(tree.longest_path())


if __name__ == '__main__':
    main()

File: cpp/template/template.cpp
/* include little library I made */
#include "../common/util.h"
#include "../common/tree.h"

auto next(ull x) -> std::vector<ull> {
    std::vector<ull> new_xs;
    return new_xs;
}

auto main() -> int {
    /* starts off with a tree with value 0 and no children and searches 20
     * iterations, feel free to tweak to whatever you want
     */
    Tree tree(0, {});
    for (int i = 0; i < 20; i++) {
        tree.step(next);
        print_vec(tree.longest_path());
    }
}
